cql3/untyped_result_set.hh

```diff
@@ -188,8 +217,19 @@ public:
         return _rows.back();
     }
 private:
+    using qr_tuple = std::tuple<foreign_ptr<lw_shared_ptr<query::result>>, shared_ptr<const cql3::metadata>>;
+    using storage = std::variant<std::monostate
+        , ::shared_ptr<cql_transport::messages::result_message>
+        , qr_tuple
+    >;
+    struct visitor;
+
+    storage _storage;
+    untyped_result_set_row::index_map_ptr _index;
     rows_type _rows;
+
     untyped_result_set() = default;
+    static untyped_result_set_row::index_map_ptr make_index(const cql3::metadata&);
```

Comment:
I know it is a matter of taste but maybe it would be better to hide this function inside .cc file and not expose it in the header since it's only used internally?

Reply:
same as above.